### ISSUE TITLE
Does chalk.js support squiggly lines?

### ISSUE THREAD

**Issue by lancejpollard** [chalk/chalk#604]
Does it support these features? https://github.com/microsoft/vscode/issues/158827

**Comment by sindresorhus** [comment]
No. It's not widely supported in terminals. Not even the plain underline style (which Chalk does support) is widely supported in terminals. We'll consider adding it, but I'd like to see some more terminal adoption first.

**Comment by rhodes-b** [comment]
I think adoption is becoming common

Foot: https://codeberg.org/dnkl/foot/pulls/1747
Ghostty: https://github.com/ghostty-org/ghostty/pull/58
Kitty: made the protocol 
Wezterm: https://github.com/wezterm/wezterm/issues/415, https://wezterm.org/faq.html#how-do-i-enable-undercurl-curly-underlines (I couldnt find relevant PR)
ITerm2: https://gitlab.com/gnachman/iterm2/-/issues/6382
VTE: https://gitlab.gnome.org/GNOME/vte/-/issues/2639

how is this considered not widely adopted? I think this covers are large userbase (Kitty, VTE terminals, iTerm2)

**Comment by Faheem-maker** [comment]
Hi,

I find this idea interesting, and I'd be happy to implement it if you're open to a PR.

**Proposed API:**

```js
console.log(chalk.squiggly.bgBlue('Hello World'));
```

Since this issue was last discussed, support for undercurl/squiggly underlines has been added to a number of major terminals, and I believe it can be useful in some of my applications.
